src/MarkdownFileCompletionItemProvider.ts

```diff
@@ -44,4 +52,17 @@ export class MarkdownFileCompletionItemProvider implements vscode.CompletionItem
         return [];
     }
   }
+
+  public async resolveCompletionItem(
+    item: MarkdownFileCompletionItem,
+    token: vscode.CancellationToken
+  ): Promise<MarkdownFileCompletionItem> {
+    const fsPath = item.fsPath;
+    if (fsPath) {
+      let note = NoteParser.noteFromFsPath(fsPath);
```

Comment:
I see below the return type of this function is `Note`, but I wonder if it should be `Note | undefined`.

Generally `NoteWorkspace._notes[fsPath]` *should* exist, but I worry there may be race conditions where if the timing of the call to this just happens to be a moment when this value has been cleared / before it has been set, we might return `undefined`

So I think we might want to modify the return type of the function and add handling to the callers (like this line) to handle the case where `note = undefined`

Reply:
It should be `Note | undefined`, I will fix it.